src/menu-items/index.ts

```diff
@@ -7,25 +7,28 @@ import VendorsService from "../vendors/vendors.service";
 const createMenuItemsModule = (
     menuItemsRepository: MenuItemsRepository,
     permissionsMiddleware: PermissionsMiddleware,
+    authorizationMiddleware: Middleware,
     vendorsService: VendorsService,
 ) => {
     const menuItemsService = new MenuItemsService(menuItemsRepository, vendorsService);
     const menuItemsController = new MenuItemsController(menuItemsService);
 
     const router = Router();
 
-    router.post('', permissionsMiddleware.isVendor, menuItemsController.createMenuItem.bind(menuItemsController));
-    router.get('/:vendorId/:menuItemId', menuItemsController.getMenuItem.bind(menuItemsController));
-    router.get('/:vendorId', menuItemsController.listMenuItemsForVendor.bind(menuItemsController));
-    router.get('', menuItemsController.listAllMenuItems.bind(menuItemsController));
+    router.post('', authorizationMiddleware, permissionsMiddleware.isVendor, menuItemsController.createMenuItem.bind(menuItemsController));
+    router.get('/:vendorId/:menuItemId', authorizationMiddleware, menuItemsController.getMenuItem.bind(menuItemsController));
+    router.get('/:vendorId', authorizationMiddleware, menuItemsController.listMenuItemsForVendor.bind(menuItemsController));
```

Comment:
@C0l0red similar to the above, the menu items are a resource on their own, should they need to be bound to the vendor?

You can always fetch for a single vendor by applying a query filter

Reply:
Yeah, it really slipped my mind while designing to take it out and turn it into a query parameter.
Fixed.